cirq/ops/named_qubit.py

```diff
@@ -33,29 +35,111 @@ def __init__(self, name: str) -> None:
     def _comparison_key(self):
         return self._comp_key
 
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def with_dimension(self, dimension: int) -> 'NamedQid':
+        return NamedQid(self._name, dimension=dimension)
+
+
+class NamedQid(_BaseNamedQid):
+    """A qid identified by name.
+
+    By default, `NamedQid` has a lexicographic order. However, numbers within
+    the name are handled correctly. So, for example, if you print a circuit
+    containing `cirq.NamedQid('qid22', dimension=3)` and
+    `cirq.NamedQid('qid3', dimension=3)`, the wire for 'qid3' will
+    correctly come before 'qid22'.
+    """
+
+    def __init__(self, name: str, dimension: int) -> None:
+        """Initializes a Named Qid with a given name and dimension
+
+            Args:
+            name: The name.
+            dimension: The dimension of the qid's Hilbert space, i.e.
+                the number of quantum levels.
+        """
+        super().__init__(name)
+        self._dimension = dimension
+        self.validate_dimension(dimension)
+
     @property
     def dimension(self) -> int:
-        return 2
+        return self._dimension
+
+    def __repr__(self) -> str:
+        return f'cirq.NamedQid({self.name!r}, dimension={self.dimension})'
 
     def __str__(self) -> str:
-        return self._name
+        return f'{self.name} (d={self.dimension})'
+
+    @staticmethod
+    def range(*args, prefix: str, dimension: int) -> List['NamedQid']:
+        """Returns a range of `NamedQids`.
+
+        The range returned starts with the prefix, and followed by a qid for
+        each number in the range, e.g.:
+
+            >>> cirq.NamedQid.range(3, prefix='a', dimension=3)
+            [cirq.NamedQid('a0', dimension=3), cirq.NamedQid('a1', dimension=3)\
+, cirq.NamedQid('a2', dimension=3)]
```

Comment:
Does it work if you only have 8 spaces here?  I'm worried this will affect the indentation of the rest of the doc string.